lib/logstash/inputs/file.rb

```diff
@@ -199,15 +199,14 @@ def run(queue)
         queue << event
       end
     end
-    finished
   end # def run
 
   public
-  def teardown
-    if @tail
-      @tail.sincedb_write
-      @tail.quit
-      @tail = nil
-    end
-  end # def teardown
+  def stop
+    @tail.quit if @tail # _sincedb_write is called implicitly
+  end
+
+  def close
+    @tail = nil
```

Comment:
This is more of a question than a critique. Isn't the entire plugin going to disappear on shutdown? Is there a practical reason for this?